PWGHF/HFC/TableProducer/HFCorrelatorD0Hadrons.cxx

```diff
@@ -0,0 +1,487 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFCorrelatorD0Hadrons.cxx
+/// \brief D0-Hadron correlator task - data-like, MC-reco and MC-kine analyses.
+///
+/// \author Samrangy Sadhu <samrangy.sadhu@cern.ch>, INFN Bari
+/// \author Swapnesh Santosh Khade <swapnesh.santosh.khade@cern.ch>, IIT Indore
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "PWGHF/DataModel/HFSecondaryVertex.h"
+#include "PWGHF/DataModel/HFCandidateSelectionTables.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Framework/runDataProcessing.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod::hf_cand;
+using namespace o2::aod::hf_cand_prong2;
+using namespace o2::aod::hf_correlation_d0hadron;
+using namespace o2::analysis::hf_cuts_d0_topik;
+using namespace o2::constants::math;
+
+///
+/// Returns deltaPhi value in range [-pi/2., 3.*pi/2], typically used for correlation studies
+///
+double getDeltaPhi(double phiD, double phiDbar)
+{
+  return RecoDecay::constrainAngle(phiDbar - phiD, -o2::constants::math::PI / 2.);
+}
+
+const int npTBinsMassAndEfficiency = o2::analysis::hf_cuts_d0_topik::npTBins;
+const double efficiencyDmesonDefault[npTBinsMassAndEfficiency] = {};
+auto efficiencyDmeson_v = std::vector<double>{efficiencyDmesonDefault, efficiencyDmesonDefault + npTBinsMassAndEfficiency};
+
+// histogram binning definition
+const int massAxisBins = 120;
+const double massAxisMin = 1.5848;
+const double massAxisMax = 2.1848;
+const int phiAxisBins = 32;
+const double phiAxisMin = 0.;
+const double phiAxisMax = 2. * o2::constants::math::PI;
+const int yAxisBins = 100;
+const double yAxisMin = -5.;
+const double yAxisMax = 5.;
+const int ptDAxisBins = 180;
+const double ptDAxisMin = 0.;
+const double ptDAxisMax = 36.;
+const double massD0 = 1.864;
+const double softPiMass = 0.14543;
+auto massPi = RecoDecay::getMassPDG(kPiPlus);
+auto massK = RecoDecay::getMassPDG(kKPlus);
```

Comment:
Why is `softPiMass` used insted of `massPi`?
Why is `massD0` not assigned using `RecoDecay::getMassPDG`?